[Thumb1] LivePhysRegs to LiveRegUnits

AZero13 opened the pull request:
This removes the r7 exception because otherwise, LiveRegUnits will try to use that register.

efriedma-quic commented on llvm/test/CodeGen/Thumb/PR35481.ll:
```diff
@@ -18,11 +18,10 @@ define <4 x i32> @f() local_unnamed_addr #0 {
 ; CHECK-V4T-NEXT:    movs r2, #3
 ; CHECK-V4T-NEXT:    movs r3, #4
 ; CHECK-V4T-NEXT:    bl g
+; CHECK-V4T-NEXT:    ldr r7, [sp, #4]
```
This doesn't look right: if we have a frame pointer in r7, we can't use r7 to store the return address.  (This matters for anything trying to asynchronously unwind using the frame pointer chain, like profilers.)

AtariDreams replied:
r7 is popped to later, so this is fine.

AtariDreams replied:
In this context, r7 is popped after it is used to restore the return address to lr.

AtariDreams replied:
Like, immediately after.

efriedma-quic commented:
Please fix the commit message (which is taken from the first comment in the thread when I hit "squash and merge").

Also, your email address is currently hidden (see https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it/74223/)

AtariDreams commented:
> Please fix the commit message (which is taken from the first comment in the thread when I hit "squash and merge").
> 
> Also, your email address is currently hidden (see https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it/74223/)

Done!

efriedma-quic commented:
Email address is still hidden.  Quoting the automated message that's supposed to trigger (not sure why it didn't trigger here):

> ⚠️ We detected that you are using a GitHub private e-mail address to contribute to the repo. Please turn off [Keep my email addresses private](https://github.com/settings/emails) setting in your account. See [LLVM Discourse](https://discourse.llvm.org/t/hidden-emails-on-github-should-we-do-something-about-it) for more information.

AZero13 committed:
[Thumb1] Switch to LiveRegUnits

This removes the r7 exception because otherwise, LiveRegUnits will try to use that register.

AtariDreams commented:
@efriedma-quic Fixed

efriedma-quic commented:
Still hidden.

AtariDreams commented:
> Still hidden.

Nope. See the commit itself.

efriedma-quic commented:
The "squash and merge" button in the GitHub UI only depends on your GitHub account settings, not the "author" line in the commit.  (I don't really want to cherry-pick the patch manually.)

AtariDreams commented:
> The "squash and merge" button in the GitHub UI only depends on your GitHub account settings, not the "author" line in the commit.  (I don't really want to cherry-pick the patch manually.)

My github account setting changed to my public email so I don't understand

AtariDreams commented:
> The "squash and merge" button in the GitHub UI only depends on your GitHub account settings, not the "author" line in the commit.  (I don't really want to cherry-pick the patch manually.)

There's only one commit so you just have to merge it.

efriedma-quic commented:
> > The "squash and merge" button in the GitHub UI only depends on your GitHub account settings, not the "author" line in the commit.  (I don't really want to cherry-pick the patch manually.)
> 
> My github account setting changed to my public email so I don't understand

There's a checkbox "Keep my email addresses private" at https://github.com/settings/emails .  If you have it checked, operations through the GitHub UI set the the author email address to "83477269+AtariDreams@users.noreply.github.com", instead of whatever you have set.

> There's only one commit so you just have to merge it.

If it was just the one time, I'd just do it, but this is going to be a repeating problem for anyone merging commits on your behalf.

AtariDreams commented:
> > > The "squash and merge" button in the GitHub UI only depends on your GitHub account settings, not the "author" line in the commit.  (I don't really want to cherry-pick the patch manually.)
> > 
> > 
> > My github account setting changed to my public email so I don't understand
> 
> There's a checkbox "Keep my email addresses private" at https://github.com/settings/emails . If you have it checked, operations through the GitHub UI set the the author email address to "[83477269+AtariDreams@users.noreply.github.com](mailto:83477269+AtariDreams@users.noreply.github.com)", instead of whatever you have set.
> 
> > There's only one commit so you just have to merge it.
> 
> If it was just the one time, I'd just do it, but this is going to be a repeating problem for anyone merging commits on your behalf.

Fixed!

AtariDreams commented:
Remove r7 exception because it messes up profilers and others.

AtariDreams commented:
/pull-request 9669aba13295de5ccdefc44e22e30c0295e6afd2
